tests/README.md

```diff
@@ -0,0 +1,207 @@
+# Substrait Test Format
+
+This document describes the format for Substrait scalar test files.
+A test file consists of the following elements:
+
+1. Version declaration
+2. Optional include statements
+3. One or more test groups, each containing one or more test cases
+
+## Syntax
+
+### Version Declaration
+The version declaration must be the first line of the file. It specifies the version of the test file format. The version declaration must be in the following format:
+```
+### SUBSTRAIT_SCALAR_TEST: V1
```

Comment:
Presumably one would choose SCALAR_TEST for a file containing only scalar tests and AGGREGATE_TEST for aggregate tests.  Might be nice to mention that (and that the files cannot have mixed test types).